src/Engine/SaveLoad.cpp

```diff
@@ -579,7 +579,7 @@ void SavegameList::Initialize() {
     }
 
     if (uNumSavegameFiles)
-        std::sort(&pSavegameList->pFileList[0], &pSavegameList->pFileList[uNumSavegameFiles - 1]);
+        std::sort(&pSavegameList->pFileList[0], &pSavegameList->pFileList[uNumSavegameFiles]);
```

Comment:
Please check. Will cause out of bounds with full save list when uNumSavegameFiles == MAX_SAVE_SLOTS.